spec/syntax.md

```diff
@@ -229,63 +210,61 @@ if it meets additional semantic requirements about its structure, defined below.
 
 ### Message
 
-A single message is either a plain message, a single pattern, or has a preamble
+A single message is either a single pattern, or has a `match` statement
 followed by one or more variants which represent the translatable body of the message.
 
 ```ebnf
-Message ::= Plain | Pattern | Preamble Variant+
+Message ::= Declaration* ( Pattern | Selector Variant+ )
 ```
 
-### Plain
+### Variable Declarations
 
-A plain message only contains translatable content;
-placeholders or their delimiters are not allowed inside a plain message.
-Plain messages must not start with one of the syntax characters `[`, `{` or `$`,
-as those would indicate that the message has a more complex structure.
-Any whitespace at the beginning or end of a plain message is ignored.
-A plain message cannot represent an empty string;
-for that, use an empty pattern `[]` instead.
+A variable declaration is an expression binding a variable identifier
+within the scope of the message to the value of an expression.
+This local variable may then be used in other expressions within the same message.
 
 ```ebnf
-Plain ::= PlainStart (PlainChar* PlainEnd)?  /* ws: explicit */
-PlainChar ::= AnyChar - ('{' | '}')
-PlainStart ::= PlainChar - ('[' | '$' | WhiteSpace)
-PlainEnd ::= PlainChar - WhiteSpace
+Declaration ::= 'let' WhiteSpace Variable '=' '{' Expression '}'
```

Comment:
`let` is clear enough, but I think there's an opportunity here to use a keyword that  directly relates to the name that we choose for these let bindings (#248). This would be self-explanatory and easier to learn and search for. For example:

    local $foo = {$count :number}   // ...and call them "locals"
    alias $foo = {$count :number}   // ...and call them "aliases"
    decl $foo = {$count :number}    // ...and call them "declarations"
    expr $foo = {$count :number}    // ...and call them "named expressions"

Reply:
Since this PR is now closed, let's use #289 to continue this topic.